drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -66,15 +63,15 @@ static int pd_intel_adsp_init(const struct device *dev)
 
 #define DT_DRV_COMPAT intel_adsp_power_domain
 
-#define POWER_DOMAIN_DEVICE(id)                                                                    \
-	static struct pg_registers pd_pg_reg##id = {                                               \
-		.wr_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRCTL_OFFSET,       \
-		.rd_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRSTS_OFFSET,       \
-		.SPA_bit = DT_INST_PROP(id, bit_position),                                         \
-		.CPA_bit = DT_INST_PROP(id, bit_position),                                         \
-	};                                                                                         \
-	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);                                     \
-	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),                   \
+#define POWER_DOMAIN_DEVICE(id)							\
+	static struct pg_registers pd_pg_reg##id = {				\
+		.wr_address = &ace_dfpmccu.dfpwrctl,				\
+		.rd_address = &ace_dfpmccu.dfpwrsts,				\
```

Comment:
Where `ace_dfpmccu` declared ? Shouldn't it be `ACE_DfPMCCU->dfpwrsts` ?

Reply:
I would remove wr_address and rd_address from here, and remove them from `struct pg_registers`.

Than change calls like:

`sys_write32(sys_read32(reg->wr_address) | SPA_bit_mask, reg->wr_address);`

to be:

sys_write32(sys_read32(ACE_DfPMCCU->dfpwrsts) | SPA_bit_mask, ACE_DfPMCCU->dfpwrsts);

(and even better some should write a helper routine that does the masking and reading)